src/config-loader/config-loader.ts

```diff
@@ -52,294 +52,267 @@ export interface Config {
 // Setup function
 export const setup = (app: _express.Application) => {
 	const loadConfig = (data: Config): Bluebird<void> =>
-		sbvrUtils.db
-			.transaction(tx =>
-				Bluebird.try(() => {
-					const authApiTx = sbvrUtils.api.Auth.clone({
-						passthrough: {
-							tx,
-							req: permissions.root,
-						},
-					});
+		sbvrUtils.db.transaction(async tx => {
+			const authApiTx = sbvrUtils.api.Auth.clone({
+				passthrough: {
+					tx,
+					req: permissions.root,
+				},
+			});
 
-					if (data.users != null) {
-						const permissionsCache: {
-							[index: string]: Bluebird<number>;
-						} = {};
-						_.each(data.users, user => {
-							if (user.permissions == null) {
-								return;
-							}
-							_.each(user.permissions, permissionName => {
-								if (permissionsCache[permissionName] != null) {
-									return;
-								}
-								permissionsCache[permissionName] = authApiTx
-									.get({
+			const { users } = data;
+			if (users != null) {
+				const permissionsCache: {
+					[index: string]: Promise<number>;
+				} = {};
+				_.each(users, user => {
+					if (user.permissions == null) {
+						return;
+					}
+					_.each(user.permissions, permissionName => {
+						if (permissionsCache[permissionName] != null) {
+							return;
+						}
+						permissionsCache[permissionName] = (async () => {
+							try {
+								const result = (await authApiTx.get({
+									resource: 'permission',
+									options: {
+										$select: 'id',
+										$filter: {
+											name: permissionName,
+										},
+									},
+								})) as Array<{ id: number }>;
+								if (result.length === 0) {
```

Comment:
How about destructuring the results as `const [result] = ...` and then early return like
```
if (result != null) {
    return result.id;
}
```
to make the code shallower (instead of checking the length)?